### PR TITLE
Feature/persist lang

### LINKED ISSUE #12: [Feature] Provide mechanism for remembering selected language
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Proposal
Based on the documentation I didn't see a way to specify a mechanism for remembering the language that user selects. I'd expect something similar to what [ngx-translate-cache](https://github.com/jgpacheco/ngx-translate-cache) provides.

## Current behavior
Proposed behavior does not exist

## Expected behavior
Pending design

## What is the motivation / use case for changing the behavior?
There is no need to ask the consumer of this library to write all the boilerplate needed to store the selected language in LocalStorage or Cookies

### PR DESCRIPTION
**Pull request by NetanelBasal**
closes #12

### CODE REVIEW

File: CREATE_LIBS.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+In development mode, you can use APIs from `ngneat/transloco` in your library, and everything will work.
+
+When you need to build the library, you'll need to do the following for the build to pass:
+
+- In the library folder run npm i `@ngneat/transloco --no-save`
+- Remove the following line from `tsconfig.lib.json`:
+  `"extends": "../../../tsconfig.json"`
+
+Now you should be able to build the library.
```
Comment (itayod) [inline CREATE_LIBS.md]: What about adding Transloco to peer dependencies?

File: projects/ngneat/transloco-persist-lang/LICENSE [hunk 2]
```diff
@@ -0,0 +1,20 @@
+The MIT License
```
Comment (itayod) [inline projects/ngneat/transloco-persist-lang/LICENSE]: do we need a license for each library? I think that one on the root is enough

File: projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { InjectionToken, Provider } from '@angular/core';
+
+export type PersistLangConfig = {
+  storage: Provider;
+  storageKey?: string;
+  getLangFn?: (langs: { cachedLang: string; browserLang: string; cultureLang: string; defaultLang: string }) => string;
+};
+
+export const TRANSLOCO_PERSIST_LANG_STROAGE = new InjectionToken('TRANSLOCO_PERSIST_LANG_STROAGE');
```
Comment (itayod) [inline projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts]: add type

File: projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+export const TRANSLOCO_PERSIST_LANG_CONFIG = new InjectionToken('TRANSLOCO_PERSIST_LANG_CONFIG');
```
Comment (itayod) [inline projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts]: same

File: projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts (hunk 3 above)
Comment (fxck) [reply projects/ngneat/transloco-persist-lang/src/lib/persist-lang.config.ts]: > STROAGE

File: projects/ngneat/transloco-persist-lang/LICENSE (hunk 2 above)
Comment (NetanelBasal) [reply projects/ngneat/transloco-persist-lang/LICENSE]: No, it's a separate package.

**Pull request merged**